### ISSUE TITLE
breakfix: unable to use the email field in an acl with oauth2 backend

### ISSUE THREAD

**Issue by btoonk** [greenpau/caddy-auth-portal#215]
**Describe the issue**

Using the auth portal and the authorization plugin I'm unable to use the email field in an acl.
I configured a generic oauth2 backend (in this case keycloak with a google backend).

The authentication works fine, after authentication the browser has an acces_token cookie with the JWT token.
Adding an acl rule with a match on the email field results in a 403 Forbidden.

It makes no difference when using mail or email as the field name.

**Configuration**

Paste full `Caddyfile` below:
```
{
  debug
}


localhost {
  route /auth* {
    authp {
      backends {
        generic_oauth2_backend {
          method oauth2
          realm generic
          provider generic
          client_id <id>
          client_secret <secret>
          scopes openid email profile
          base_auth_url https://<keycloak-url>
          metadata_url https://<keycloak-metadata-url>
        }
      }
    }
  }

  route {
    authorize {
      primary yes
      set auth url /auth/oauth2/generic
      acl rule {
        comment allow email
        suffix match email "@<domain>"
        allow stop log debug
      }
    }
    reverse_proxy https://httpbin.org
  }
}
```

**Version Information**

Provide output of `caddy list-modules -versions | grep auth` below:

```
http.authentication.hashes.bcrypt v2.4.6
http.authentication.hashes.scrypt v2.4.6
http.authentication.providers.http_basic v2.4.6
http.handlers.authentication v2.4.6
http.authentication.providers.authorize v1.3.22
http.handlers.authp v1.4.36
```

**Expected behavior**

Matching on the email or mail field in an acl would allow the user to access this resource. 

**Additional context**

The field name used in the JWT token for the mail address is `mail`.
https://github.com/greenpau/caddy-authorize/blob/main/pkg/user/user.go#L543-L544

But the code used for matching the mail address is always using the field `email` (also if you specify `mail` in the acl).
https://github.com/greenpau/caddy-authorize/blob/main/pkg/acl/condition.go#L52

I think this is the reason why this is not working.

**Comment by btoonk** [comment]
A possible solution that works for me is changing the mail address field in the JWT token to email or to the name it originally was. I can create a PR if you agree with this solution.

```
diff --git a/pkg/user/user.go b/pkg/user/user.go
index cc6bd49..97395fc 100644
--- a/pkg/user/user.go
+++ b/pkg/user/user.go
@@ -540,8 +540,8 @@ func (c *Claims) unpackMail(k string, v interface{}, mkv, tkv map[string]interfa
        default:
                return errors.ErrInvalidEmailClaimType.WithArgs(k, v)
        }
-       tkv["mail"] = c.Email
-       mkv["mail"] = c.Email
+       tkv[k] = c.Email
+       mkv[k] = c.Email
        return nil
 }
```

**Comment by greenpau** [comment]
@btoonk , have you tried changing `email` to `mail`?

```
      acl rule {
        comment allow email
        suffix match mail "@<domain>"
        allow stop log debug
      }
```

**Comment by greenpau** [comment]
@btoonk , on a separate note, please see https://github.com/greenpau/caddy-auth-portal/issues/216. I figured if you got to a point of the above diff, you could challenge yourself with making your config smaller. Please let me know if you want to engage on it.

**Comment by btoonk** [comment]
@greenpau changing the field from `email` to `mail` does not make any difference.
At the moment I don't use caddy, but if we decide to do so I definitely want to help to make this config smaller.

**Comment by greenpau** [comment]
> changing the field from email to mail does not make any difference.

@btoonk , got it! Will dig into it this week.

**Comment by greenpau** [comment]
@btoonk , I applied the fix in [v1.4.37](https://github.com/greenpau/caddy-auth-portal/releases/tag/v1.4.37). You should be able to apply the following ACL if you are using `github.com/greenpau/caddy-authorize v1.3.24`.

```
            acl rule {
                suffix match mail "@localdomain.local"
                allow stop log debug
            }
            acl default deny
```

Please test 😃

**Comment by btoonk** [comment]
@greenpau Well done! This issue is now solved.
ACLs that match on `mail` or` email` now work fine.
